Changed Yara service to use holmeslibrary

ms-xy opened the pull request:
Made the yara service ready to support the upcoming holmeslibrary changes.

ms-xy committed:
Added some basic error class for service request errors, a result set class and a class for memory efficient file access.

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy committed:
Merge branch 'master' into library-extension

ms-xy committed:
Improvements to holmeslibrary.Meta and ServiceRequestError.

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy committed:
Merge remote-tracking branch 'upstream/master' into library-extension-symlinktest

ms-xy committed:
Adjust to the limitations of Docker to be able to use the library file.

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy committed:
Yara service changed to use the holmeslibrary.

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy committed:
Slight typo in service.meta

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy committed:
Merged config and metadata

ms-xy committed:
Merge branch 'zipmeta-fix' into yara-holmeslibrary

ms-xy committed:
Slight fix, changed info and adapted url handlers.

- git checkout adapted
- bugfix for the case that yara.load is not defined (but load_rules is in that case)
- changed info html output
- unified service config and meta data
- changed url handlers to use the urls specified in the config

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy committed:
Wrong name for holmeslibrary directory on git clone fixed

Signed-off-by: Maximilian Schott <ms@roottec.com>

ms-xy committed:
Dockerfile used wrong workdir, previous bugfix removed as it applied to an outdated yara version

Signed-off-by: Maximilian Schott <ms@roottec.com>

webstergd commented:
Could you also adjust yara per the comment in PEID. i.e. have get rules read the rules config location veres the main worker.

ms-xy committed:
getrules.py would behave differently than yara_worker.py regarding the rules file path.

Removed the settings.YaraRules setting in favor of the yara_rules.local_path setting.
Further made both python files respect this setting.

ms-xy commented:
updated as per request, as well as #72
